store/tikv/rawkv.go

```diff
@@ -298,3 +322,106 @@ func (c *RawKVClient) sendDeleteRangeReq(startKey []byte, endKey []byte) (*tikvr
 		return resp, actualEndKey, nil
 	}
 }
+
+func (c *RawKVClient) sendBatchPut(bo *Backoffer, keys, values [][]byte) error {
+	keyToValue := make(map[string][]byte)
+	for i, key := range keys {
+		keyToValue[string(key)] = values[i]
+	}
+	groups, _, err := c.regionCache.GroupKeysByRegion(bo, keys)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	var batches []batch
+	// split the keys by size and RegionVerID
+	for regionID, groupKeys := range groups {
+		batches = appendBatches(batches, regionID, groupKeys, keyToValue, rawBatchPutSize)
+	}
+	bo, cancel := bo.Fork()
+	ch := make(chan error, len(batches))
+	for _, batch := range batches {
+		rawKVClientGP.Go(func() {
+			singleBatchBackoffer, singleBatchCancel := bo.Fork()
+			defer singleBatchCancel()
+			ch <- c.doBatchPut(singleBatchBackoffer, batch)
+		})
+	}
+	err = nil
```

Comment:
I suppose there is no need to set it. `err` is guaranteed to be nil here.

Reply:
Done